### PR TITLE
feature: add keycard metric events

### LINKED ISSUE #21685: Keycard - Add analytics
# Problem

We should have a clearer picture about the success criteria for Keycard integration in Status.

## Acceptance Criteria

- Verify if we are tracking keycard flows via navigation events in the analytics tool.
- Verify if the navigation events for keycard flows have clear and precise names (even for analytics users without knowledge about the code), otherwise consider pushing new events when certain flows end (e.g. when profile migration to keycard finishes successfully).
- Update our current source of truth in Notion doc https://www.notion.so/Event-Tracking-Specification-Mobile-11a8f96fb65c8032a05efcebea3a1ce6#11a8f96fb65c802d9380e2de220e391e

## Out of scope

This issue is only catering for Keycard features being integrated by the Mobile Core team, not the Wallet team. cc @shivekkhurana

### PR DESCRIPTION
**Pull request by seanstrom**
fixes https://github.com/status-im/status-mobile/issues/21685

### Summary

* This PR attempts to add metric events for the following keycard events:
  * When a user successfully logins with Keycard
  * When a user succeeds or fails to migrate profile to Keycard
  * When a Keycard user encounters a feature that is unavailable
  * When a Keycard user navigates to the feature upvote url for an unavailable feature

#### Platforms

- Android
- iOS

##### Functional

- Keycard analytics

### Steps to test

- WIP

status: ready

### CODE REVIEW

File: src/status_im/contexts/keycard/migrate/re_encrypting/events.cljs [hunk 1]
```diff
@@ -27,5 +27,8 @@
              :callback
              (fn [{:keys [error]}]
                (if (string/blank? error)
-                 (rf/dispatch [:navigate-to :screen/keycard.migrate.success])
-                 (rf/dispatch [:navigate-to :screen/keycard.migrate.fail])))}]]})))
+                 (do (rf/dispatch [:navigate-to :screen/keycard.migrate.success])
+                     (rf/dispatch [:centralized-metrics/track :metric/keycard-migration-succeeded]))
```
Comment (ilmotta) [inline src/status_im/contexts/keycard/migrate/re_encrypting/events.cljs]: I did something similar recently, but opted to track the aliased navigation event in `tracking/track-view-id-event`. No right or wrong.

https://github.com/status-im/status-mobile/blob/949e78a17e7a7a9367ab264b09b34c0b88d51df9/src/status_im/contexts/centralized_metrics/tracking.cljs#L55-L56

**Review by ilmotta — APPROVED** [review]
Nice solution to track the feature name. I didn't verify if events are being sent correctly to MixPanel, but the code looks good and safe.

**Review by mohsen-ghafouri — APPROVED** [review]
👌

**Comment by seanstrom** [comment]
@ilmotta tested the new events locally with the test environment for mixpanel. Seems to be working 🙌, but feel free to check the recent event history in the mixpanel project for a quick double check.

**Comment by churik** [comment]
Please, let me know when it is merged and notify in [release-coordination channel ](https://discord.com/channels/1210237582470807632/1217172571917647932) to cherry-pick it, thanks!

**Pull request merged**